streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -611,65 +624,164 @@ public StateStore getStateStore(final String name) {
     }
 
     /**
-     * Get the {@link KeyValueStore} with the given name.
+     * Get the {@link KeyValueStore} or {@link TimestampedKeyValueStore} with the given name.
      * The store can be a "regular" or global store.
      * <p>
+     * If the registered store is a {@link TimestampedKeyValueStore} this method will return a value-only query
+     * interface. For full query access use {@link #getTimestampedKeyValueStore(String)}.
+     * <p>
      * This is often useful in test cases to pre-populate the store before the test case instructs the topology to
      * {@link #pipeInput(ConsumerRecord) process an input message}, and/or to check the store afterward.
      *
      * @param name the name of the store
-     * @return the key value store, or {@code null} if no {@link KeyValueStore} has been registered with the given name
+     * @return the key value store, or {@code null} if no {@link KeyValueStore} or {@link TimestampedKeyValueStore}
+     * has been registered with the given name
      * @see #getAllStateStores()
      * @see #getStateStore(String)
+     * @see #getTimestampedKeyValueStore(String)
      * @see #getWindowStore(String)
+     * @see #getTimestampedWindowStore(String)
      * @see #getSessionStore(String)
+     * @see #getTimestampedSessionStore(String)
      */
     @SuppressWarnings({"unchecked", "WeakerAccess"})
     public <K, V> KeyValueStore<K, V> getKeyValueStore(final String name) {
         final StateStore store = getStateStore(name);
+        if (store instanceof TimestampedKeyValueStore) {
+            return new KeyValueStoreFacade((TimestampedKeyValueStore<Object, Object>) store);
```

Comment:
I don't think it matters too much due to erasure and all the casting, but I think the reason you were prompted to use `<Object,Object>` instead of `<K,V>` as the parameters is that you forgot the generic parameters on `KeyValueStoreFacade` (so they are both expanded to `Object`). I'm a little surprised there was no warning for the raw-type return value.

I didn't test this, but I think this is what you wanted:
```suggestion
            return new KeyValueStoreFacade<K, V>((TimestampedKeyValueStore<K, V>) store);
```

(also below)

Reply:
Ack. You suggestion is almost correct. It should be:
```
return new KeyValueStoreFacade<>((TimestampedKeyValueStore<K, V>) store);
```